src/Compilers/CSharp/Test/Semantic/Semantics/CollectionLiteralTests.cs

```diff
@@ -266,577 +423,600 @@ static void Main()
-                        foreach (var x in a)
-                        {
-                            Console.Write("Length={0}, ", x.Length);
-                            foreach (var y in x)
-                                Console.Write("{0}, ", y);
-                        }
-                        Console.WriteLine();
+                        var x = F1([]);
+                        var y = F2([1, 2]);
+                        x.Report(includeType: true);
+                        y.Report(includeType: true);
                     }
                 }
                 """;
-            var verifier = CompileAndVerify(source, expectedOutput: """
-                Length=0, 
-                Length=1, Length=0, 
-                Length=2, Length=1, 1, Length=2, 2, 3, 
-                """);
-            verifier.VerifyIL("Program.Create1", """
-                {
-                  // Code size        7 (0x7)
-                  .maxstack  1
-                  IL_0000:  ldc.i4.0
-                  IL_0001:  newarr     "int[]"
-                  IL_0006:  ret
-                }
-                """);
-            verifier.VerifyIL("Program.Create2", """
-                {
-                  // Code size       16 (0x10)
-                  .maxstack  4
-                  IL_0000:  ldc.i4.1
-                  IL_0001:  newarr     "object[]"
-                  IL_0006:  dup
-                  IL_0007:  ldc.i4.0
-                  IL_0008:  ldc.i4.0
-                  IL_0009:  newarr     "object"
-                  IL_000e:  stelem.ref
-                  IL_000f:  ret
-                }
-                """);
-            verifier.VerifyIL("Program.Create3", """
-                {
-                  // Code size       52 (0x34)
-                  .maxstack  7
-                  IL_0000:  ldc.i4.2
-                  IL_0001:  newarr     "object[]"
-                  IL_0006:  dup
-                  IL_0007:  ldc.i4.0
-                  IL_0008:  ldc.i4.1
-                  IL_0009:  newarr     "object"
-                  IL_000e:  dup
-                  IL_000f:  ldc.i4.0
-                  IL_0010:  ldc.i4.1
-                  IL_0011:  box        "int"
-                  IL_0016:  stelem.ref
-                  IL_0017:  stelem.ref
-                  IL_0018:  dup
-                  IL_0019:  ldc.i4.1
-                  IL_001a:  ldc.i4.2
-                  IL_001b:  newarr     "object"
-                  IL_0020:  dup
-                  IL_0021:  ldc.i4.0
-                  IL_0022:  ldc.i4.2
-                  IL_0023:  box        "int"
-                  IL_0028:  stelem.ref
-                  IL_0029:  dup
-                  IL_002a:  ldc.i4.1
-                  IL_002b:  ldc.i4.3
-                  IL_002c:  box        "int"
-                  IL_0031:  stelem.ref
-                  IL_0032:  stelem.ref
-                  IL_0033:  ret
-                }
-                """);
+            // PROTOTYPE: Should these calls be ambiguous?
```

Comment:
Sounds reasonable. That might require that _better function member_ prefers an array over an interface (or a concrete type over an interface as in the next test) because both parameter types are considered _constructible_. I've removed the PROTOTOYPE comments.